packages/react-relay/classic/store/readRelayQueryData.js

```diff
@@ -239,11 +240,13 @@ class RelayStoreReader extends RelayQueryVisitor<State> {
           seenDataIDs: state.seenDataIDs,
           storeDataID: dataID,
         });
+        // remove null plural fields
+        if (!nextState.data) {return;}
         node.getChildren().forEach(child => this.visit(child, nextState));
         if (nextState.isPartial) {
           state.isPartial = true;
         }
-        return nextState.data;
+        nextData.push(nextState.data);
```

Comment:
Doing `foo.map(...).filter(...)` allocates two arrays instead of one: the intermediate result of map is thrown away. Functional APIs are great for readability, but unfortunately they don't have the optimizations applied that make them performant in other languages (often called fusion). We avoid this style throughout the codebase for performance reasons.